src/OpenDB/src/lefout/lefout.cpp

```diff
@@ -114,29 +213,110 @@ void lefout::writeHeader(dbLib* lib)
+  fprintf(_out, "    DATABASE MICRONS %d ;\n", database_units);
+  fprintf(_out, "END UNITS\n");
+}
+
+void lefout::writeBusBitChars(char left_bus_delimeter, char right_bus_delimeter)
+{
   fprintf(_out,
           "BUSBITCHARS \"%c%c\" ;\n",
           left_bus_delimeter,
           right_bus_delimeter);
-  fprintf(_out, "DIVIDERCHAR \"%c\" ;\n", hier_delimeter);
-  writePropertyDefinitions(lib);
+}
 
-  if (lib->getLefUnits()) {
-    fprintf(_out, "UNITS\n");
-    fprintf(_out, "    DATABASE MICRONS %d ;\n", lib->getLefUnits());
-    fprintf(_out, "END UNITS\n");
+void lefout::writeNameCaseSensitive(const dbOnOffType on_off_type)
+{
+  fprintf(_out, "NAMESCASESENSITIVE %s ;\n", on_off_type.getString());
+}
+
+void lefout::writeBlock(dbBlock* db_block)
+{
+  dbBox* bounding_box = db_block->getBBox();
+  double origin_x = lefdist(bounding_box->xMin());
+  double origin_y = lefdist(bounding_box->xMin());
+  double size_x = lefdist(bounding_box->getDX());
+  double size_y = lefdist(bounding_box->getDY());
+
+  fprintf(_out, "MACRO %s\n", db_block->getName().c_str());
+  fprintf(_out, "  FOREIGN %s 0 0 ;\n", db_block->getName().c_str());
+  fprintf(_out, "  CLASS BLOCK ;\n");
+  fprintf(_out, "  ORIGIN %g %g ;\n", origin_x, origin_y);
+  fprintf(_out, "  SIZE %g %g ;\n", size_x, size_y);
+  writePins(db_block);
+  writeObstructions(db_block);
+  fprintf(_out, "END %s\n", db_block->getName().c_str());
+}
+
+void lefout::writePins(dbBlock* db_block)
+{
+  writePowerPins(db_block);
+  writeBlockTerms(db_block);
+}
+
+void lefout::writeBlockTerms(dbBlock* db_block)
+{
+  for (dbBTerm* b_term : db_block->getBTerms()) {
+    fprintf(_out, "  PIN %s\n", b_term->getName().c_str());
+    fprintf(_out, "    DIRECTION %s ;\n", b_term->getIoType().getString());
+    fprintf(_out, "    USE %s ;\n", b_term->getSigType().getString());
+    for (dbBPin* db_b_pin : b_term->getBPins()) {
+      fprintf(_out, "    PORT\n");
+      dbSet<dbBox> term_pins = db_b_pin->getBoxes();
+      writeBoxes(term_pins, "      ");
+      fprintf(_out, "    END\n");
+    }
+    fprintf(_out, "  END %s\n", b_term->getName().c_str());
+  }
+}
+
+void lefout::writePowerPins(dbBlock* db_block)
+{  // Power Ground.
+  for (dbNet* net : db_block->getNets()) {
+    if (isPowerNet(net->getSigType())) {
+      continue;
+    }
+    fprintf(_out, "  PIN %s\n", net->getName().c_str());
+    fprintf(_out, "    USE %s ;\n", net->getSigType().getString());
+    fprintf(
+        _out, "    DIRECTION %s ;\n", dbIoType(dbIoType::INOUT).getString());
+    for (dbSWire* special_wire : net->getSWires()) {
+      fprintf(_out, "    PORT\n");
+      dbSet<dbSBox> wires = special_wire->getWires();
+      writeBoxes(wires, /*indent=*/"      ");
+      fprintf(_out, "    END\n");
+    }
+    fprintf(_out, "  END %s\n", net->getName().c_str());
   }
 }
 
+bool lefout::isPowerNet(const dbSigType& db_sig_type) const
+{
+  return db_sig_type != dbSigType::POWER && db_sig_type != dbSigType::GROUND;
```

Comment:
`db_sig_type == dbSigType::POWER || db_sig_type == dbSigType::GROUND`

Reply:
As Matt mentioned, `net->getSigType()->isSupply()` is better.